extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java

```diff
@@ -420,8 +420,10 @@ private static void populateTypicalProperty(MethodCreator methodCreator, ResultH
             createWriteValue(methodCreator, configObject, field, setter, useFieldAccess, setterValue);
 
         }
-        configPropertyBuildItemCandidates
-                .add(new ConfigPropertyBuildItemCandidate(field.name(), fullConfigName, fieldType));
+        if (field.type().kind() != Type.Kind.PRIMITIVE) { // the JVM assigns primitive types a default even though it doesn't show up in the bytecode
+            configPropertyBuildItemCandidates
+                    .add(new ConfigPropertyBuildItemCandidate(field.name(), fullConfigName, fieldType));
```

Comment:
Hmmm, and we are sure that nothing that `ConfigPropertyBuildItemCandidate` triggers will ever be needed by primitives?

Reply:
Yup, that list is only used in turn to populate `ConfigPropertyBuildItem` (which are the user properties that are required when booting the application).
The `getPropertyType` field of `ConfigPropertyBuildItem` is only in order to register for reflection which if of course is not needed for primitives.
So there is absolutely no reason that these primites should ever end up in the candidates list.

Reply:
OK, thanks for the quick turnaround.

Reply:
You are welcome!